### PR TITLE
Execute gRPC request processing on duplicated contexts

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

File: extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/services/AssertHelper.java [hunk 1]
```diff
@@ -17,6 +20,16 @@ public static void assertRunOnEventLoop() {
...
     }
 
+    public static Context assertRunOnDuplicatedContext() {
+        assertThat(Vertx.currentContext()).isNotNull();
+        assertThat(isRootContext(Vertx.currentContext())).isFalse();
+        return Vertx.currentContext();
+    }
+
+    private static boolean isRootContext(Context context) {
+        return context instanceof EventLoopContext || context instanceof WorkerContext;
```
Comment (michalszynkiewicz) [inline extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/services/AssertHelper.java:30]: Would it be good to have this method in some globally accessible utility class?

Comment (cescoffier) [reply extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/services/AssertHelper.java:30]: We are working on that. It should land in a smallrye common module (need the same in various smallrye project)

File: extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/services/BlockingMutinyTestService.java [hunk 2]
```diff
@@ -32,13 +36,15 @@
     public Uni<EmptyProtos.Empty> emptyCallBlocking(EmptyProtos.Empty request) {
         assertThat(request).isNotNull();
         assertRunOnWorker();
+        assertRunOnDuplicatedContext();
```
Comment (michalszynkiewicz) [inline extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/services/BlockingMutinyTestService.java:39]: I'm a bit surprised that not being on the event loop we still have a vertx context

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/blocking/BlockingServerInterceptor.java [hunk 3]
```diff
@@ -58,7 +58,7 @@ public Boolean apply(String name) {
 
         // We need to check if the method is annotated with @Blocking.
         // Unfortunately, we can't have the Java method object, we can only have the gRPC full method name.
-        // We extract the method name, and check if the name if is the list.
+        // We extract the method name, and check if the name if it is the list.
```
Comment (michalszynkiewicz) [inline extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/blocking/BlockingServerInterceptor.java:61]:
```suggestion
        // We extract the method name, and check if the name is in the list.
```

File: extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/services/BlockingMutinyTestService.java (hunk 2 above)
Comment (cescoffier) [reply extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/services/BlockingMutinyTestService.java:39]: Yes, I know it's confusing. But inside an "executeBlocking" block, you still have access to the same context as outside (leading to fun concurrency issue)

Comment (michalszynkiewicz) [reply extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/services/BlockingMutinyTestService.java:39]: does it work for the root contexts too? Or only duplicated ones?

**Review by michalszynkiewicz — APPROVED** [review]

**Pull request merged**